### PR TITLE
Conditionally download updates.

### LINKED ISSUE #111: Should only download an update once
If the check interval ticks over multiple times, ensure that the same update isn’t downloaded and prepared both times. It would be a waste of bandwidth to download the same update twice.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #111

### CODE REVIEW

**Comment by joshaber** [comment]
This assumes we can trust servers to return and respect etags. Is that a reasonable assumption these days?

**Comment by joshaber** [comment]
Looks like S3 respects etags which I gotta think is where most updates live.

File: Squirrel/SQRLUpdater.m [hunk 1]
```diff
@@ -40,6 +40,10 @@ @interface SQRLUpdater ()
 
 @property (atomic, readwrite) SQRLUpdaterState state;
 
+/// The etag of the currently downloaded update, nil if no update has been
+/// downloaded.
+@property (atomic, copy) NSString *etag;
```
Comment (keithduncan) [inline Squirrel/SQRLUpdater.m]: Does this need to be atomic, inside the scope of the `RACCommand` shouldn’t access be serialised?

**Comment by keithduncan** [comment]
I don’t think we lose anything with the `atomic` and I don’t want to split hairs, :+1: to merge this

File: Squirrel/SQRLUpdater.m (hunk 1 above)
Comment (joshaber) [inline Squirrel/SQRLUpdater.m]: LOL I honestly couldn't recall anymore, so I went with safe over sorry.

**Pull request merged**